test/spec/features/transparent-video.spec.js

```diff
@@ -0,0 +1,226 @@
+let myDescribe = describe;
+
+if (/phantom|HeadlessChrome|HeadlessFirefox|Chrome/i.test(navigator.userAgent)) {
+  console.warn("Transparent video skipped - Running only on Firefox");
+  myDescribe = xdescribe;
+}
+
+myDescribe("Transparent Video Test", function () {
+  let restoreXHR = () => {};
+  let timeout = 60000;
+  let cl;
+  beforeEach(() => {
+    cl = new cloudinary.Cloudinary({cloud_name: "sdk-test"});
+    createTestContainer();
+  });
+  afterEach(() => {
+    removeSeeThruScript();
+    removeTestContainer();
+    restoreXHR();
+  });
+
+  // Annotations, these will be removed before we merge.
+  // TODO global
+  //  DONE - Skip this test in Chrome and Headless - Run only in FireFox (There was an issue with Chrome in tests)
+
+  // TODO - seeThru - What was tested?
+  //  DONE - Ensure custom class is passed successfully to Canvas
+  //  DONE - test max_timout actually times out
+  //  DONE - test that we already have seeThru (load a URL twice, ensure no double scriptTags)
+  //  DONE - test loop parameter exists
+  //  DONE - test for seeThru URL
+
+  // TODO NativeSupport
+  //  DONE - Ensure custom class is passed successfully to video element
+  //  DONE - test max_timout
+  //  DONE - test loop parameter exists
+  //  DONE - Test our extra arguments are not passing to video element
+
+  describe('SeeThru tests', () => {
+    it("Display seeThru canvas if there is no Native support for video", function (done) {
+      restoreXHR = forceNativeTransparentSupport(false);
+      let container = getTestContainer();
+
+
+      cl.createTransparentVideo(container, 'transparentVideoTests/transparent-girl', {
+        loop: true,
+        max_timeout: timeout,
+        class: 'a-custom-class'
+      }).then((res) => {
+        let canvas = res.querySelector('.cld-transparent-video');
+        let video = res.querySelector('video');
+
+        expect(canvas).not.toBe(null);
+        expect(res).toBe(container);
+        expect(video.hasAttribute('autoplay')).toBe(true);
+        expect(video.hasAttribute('loop')).toBe(true);
+        expect(video.hasAttribute('muted')).toBe(true);
+        expect(video.getAttribute('src')).toContain('blob:http://');
+        expect(video.hasAttribute('seethruurl')).toBe(false);
+        expect(video.hasAttribute('max_timeout')).toBe(false);
+
+        let classes = canvas.className.split(/\s+/);
+        expect(classes.indexOf('a-custom-class')).toBeGreaterThanOrEqual(0);
+        done();
+      }).catch((err) => {
+        // Fail test if we reach the catch
+        expect(err).toBeUndefined();
+        done();
+      });
+    }, timeout); // timeout
+
+    it("Should timeout with a short enough max_timeout", function (done) {
```

Comment:
it really is that generic,
The entire process should fail if timeout is short enough, I can't really guarantee what exactly will fail or how far the system will reach in 1ms, but it will 100% not finish.

most likely that in 1ms, we won't get through the script fetch for seeThru, but there's no guarantee (it could be cached for example, and then it'll be instant).